### ISSUE TITLE
Changing non-root user name from 'vscode' to something more general

### ISSUE THREAD

**Issue by natescherer** [devcontainers/images#145]
Hello,
I was just curious if there were any plans to change the name of the non-root user in images from being 'vscode' to something more general given the move away from being part of vscode.

While I recognize it would probably be a fairly major breaking change, I feel like it would be the right move going forward.

Thanks!

**Comment by samruddhikhandale** [comment]
Hi @natescherer 👋 

Yes, we had a discussion about renaming the `vscode` user when we were in the process of migration. We intentionally didn't change then as migration itself was a big change, and we didn't want to confuse folks. However, I believe we could soon pick this up. // cc @Chuxel / @bamurtaugh  @devcontainers/maintainers 

 Unfortunately, I don't have an issue of hand for the discussion, we could now use this one for tracking the re-naming. Thanks! 

> While I recognize it would probably be a fairly major breaking change, I feel like it would be the right move going forward.

Yes, I agree 💯 . We had discussed some remedies to contain the problems associated with this change. ( eg. bumping major versions, adding back compatibility/ error handling with prominent messaging). We'd have a concrete plan once someone picks this up.

**Comment by Chuxel** [comment]
Yes, the only reason it hasn't been done is due to the breaking change for existing image users.

**Comment by jkeech** [comment]
To summarize an offline discussion we had about this issue:
* We do want to change the user for the images from `vscode` to something generic, like `devcontainer`
* We need to plan how to do it carefully since it is a breaking change, and we want to minimize the impact to repos with existing devcontainer configurations that use the `vscode` user
* It will be a major version bump on the images, but we also want to see if there's a way to more gracefully roll out this change
* One thing that's related which we can do is to improve the error messaging if the specified `remoteUser` does not exist in the container image. Today users get hard-to-intepret error messages when there's a mismatch. If we change the user in the container and users still have `"remoteUser":"vscode"`, we'd want to at least have a clear error message.

**Comment by natescherer** [comment]
All sounds good to me.

Major version bump should insulate anyone who's using semver properly but I'll freely admit to lazily using the latest tag too often, ha.

Perhaps in advance of the change you could have the containers spit out warning messages similar to how Azure CLI does in advance of major breaking changes?

**Comment by hoechenberger** [comment]
Hello, I wanted to ask if any progress has been made on this issue since last year? :) Thank you!

**Comment by thclark** [comment]
> * We do want to change the user for the images from `vscode` to something generic, like `devcontainer`

I think something like `developer` or `devuser` might be less confusing, since `devcontainer` refers to the actual container rather than the user within it.

But a huge thumbs up for this.

On the subject of breaking changes, it's honestly very common for changes here to break stuff downstream. We just have to deal with it as developers consuming this. The main problem is that quite a few features/scripts scattered around simply hard code the `vscode` or `/home/vscode` paths, but honestly where we've done that (yes, I'm one of them) it's our funeral. 

To help mitigate, you could potentially add the new `devuser` alongside  a `vscode` user, shift all the documentation over, give it a while to percolate then deprecate the `vscode` user, at least people will have had time to adjust?
